extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/TransactionManagerConfiguration.java

```diff
@@ -19,8 +16,92 @@
+     * mechanism: RecoveryEnvironmentBean, JTAEnvironmentBean, CoreEnvironmentBean, CoordinatorEnvironmentBean
+     * and ObjectStoreEnvironmentBean.
+     * </p>
+     * <p>
+     * To enable full access to any config option provided by these environment beans set this quarkus
+     * configuration item to true. The extension will then check whether or not there are any java
+     * system properties of the form "environment bean name"."property name" and if present will use
+     * that value to configure the transaction manager.
+     * </p>
+     * <p>
+     * If any Narayana transaction manager config property is not set via a system property then
+     * the extension will assume the following default values for the bean properties:
+     * </p>
+     * <p>
+     * <table style="width:100%">
+     * <tr>
+     * <th>System Property</th>
+     * <th>Value</th>
+     * </tr>
+     * <tr>
+     * <td>JTAEnvironmentBean.xaResourceOrphanFilterClassNames</td>
+     * <td>com.arjuna.ats.internal.jta.recovery.arjunacore.JTATransactionLogXAResourceOrphanFilter,
+     * com.arjuna.ats.internal.jta.recovery.arjunacore.JTANodeNameXAResourceOrphanFilter,
+     * com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinateJTAXAResourceOrphanFilter,
+     * com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinationManagerXAResourceOrphanFilter
+     * </td>
+     * </tr>
+     * <tr>
+     * <td>RecoveryEnvironmentBean.recoveryModuleClassNames</td>
+     * <td>com.arjuna.ats.internal.arjuna.recovery.AtomicActionRecoveryModule,
+     * com.arjuna.ats.internal.txoj.recovery.TORecoveryModule,
+     * com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinateAtomicActionRecoveryModule,
+     * com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule
+     * com.arjuna.ats.internal.jta.recovery.arjunacore.CommitMarkableResourceRecordRecoveryModule
+     * </td>
+     * </tr>
+     * <tr>
+     * <td>RecoveryEnvironmentBean.expiryScannerClassNames</td>
+     * <td>com.arjuna.ats.internal.arjuna.recovery.ExpiredTransactionStatusManagerScanner
+     * </td>
+     * <tr>
+     * <td>RecoveryEnvironmentBean.recoveryPort</td>
+     * <td>4712</td>
+     * </tr>
+     * <tr>
+     * <td>RecoveryEnvironmentBean.recoveryListener</td>
+     * <td>true</td>
+     * </tr>
+     * </table>
+     * </p>
+     * <p>
+     * The following two Narayana config properties cannot be set in this way:
+     * </p>
+     * <p>
+     * <em>JTAEnvironmentBean.xaRecoveryNodes</em> which is always initialised to the value of
+     * {@link TransactionManagerConfiguration#nodeName}
+     * </p>
+     * <p>
+     * <em></em>CoordinatorEnvironmentBean.transactionStatusManagerEnable</em> which is always initialised to the value false
+     * </p>
+     * <p>
+     * Note that quarkus config properties will override Narayana system properties.
+     * </p>
+     */
+    @ConfigItem(name = "checkForNarayanaSystemProperties", defaultValue = "false")
```

Comment:
isn't this approach a bit in contrast (or in parallel) to how the Quarkus works with properties normally?

Reply:
The goal of allowing system properties is to provide a mechanism to configure an option that isn't exposed in the quarkus config model. It is not the intention to make everything configurable via the quarkus model.

The intention is to gradually expose certain behaviours as we get requests for them. For example, in a subsequent issue I will be adding quarkus properties to configure the journal and JDBC stores. But providing access to the Narayana property mechanism means that these and other features can be enabled now if desired.

Reply:
You don't need the `name` or `defaultValue` on this one (the property name would then be `check-for-narayana-system-properties` though, in line with our normal naming convention).

Reply:
To address @ochaloup's concern: if we do allow this, we'd want to make a note that this option is specifically for advanced users only.  The problem is that users are going to rely on this capability if we expose it.  Our established practice in situations such as this one has been to let users complain/request a feature which we then explicitly expose via configuration rather than support any sort of pass-through.

Reply:
Yes, @dmlloyd expressed exactly what I was in doubt about.

Plus, if the goal would be to gradually expose certain behaviours then the properties handling needs to be much more complicated than it's in this PR. What's the part of the PR ((https://github.com/quarkusio/quarkus/blob/95ef7f637da601458273eb8625f67d6baaff7256/extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/NarayanaJtaRecorder.java#L87) you may use either the "quarkus property system" or "narayana system property system". For the gradual evolution, I assume, there should be the rule that a defined quarkus property overrides any used system property.

Reply:
The PR already states the convention. It is present in the javadoc that I created, ie:

```java
     * Note that quarkus config properties will override Narayana system properties.
```

But I am still of the opinion that we should evolve which set of properties to expose via the quarkus config system. Without this mechanism users will have to wait for us to add the feature.

Reply:
ie the javadoc for the system property override states how the user can enable Narayana features. But note that the user can already configure Narayana using XML property files (but this PR removes that capability) so not having the new option that this PR introduces is removing functionality.

Reply:
> But I am still of the opinion that we should evolve which set of properties to expose via the quarkus config system. Without this mechanism users will have to wait for us to add the feature.

That is exactly the intent.  We've done all other extensions this way as well.

> But [...] not having the new option that this PR introduces is removing functionality.

Right, we are deliberately avoiding exposing every bit of functionality with extensions (not just this one but all of them); rather we put a heavy preference towards only exposing the most commonly used configuration items, at least to start with.  A user should have to configure few or no properties in order to get a working, useful system, and having too many knobs can be an attractive nuisance.  As users need additional control, we can easily add configuration properties to expose what is needed (which have a low implementation complexity, generally speaking).

Reply:
> > But [...] not having the new option that this PR introduces is removing functionality.
> al control, we can easily add configuration properties to expose what is needed (which have a low implementation complexity, generally speaking).

My point was that with the existing extension it is possible to provide a jbossts-properties.xml config file. Do you still want me to remove that capability because if I remove that capability and don't allow the use of system properties then I will be introducing a regression?

Reply:
We can't remove functionality which existed in 1.0.  I would just avoid the system property layer and leave the XML in.  But that's a recommendation, not a hard requirement.

Reply:
The main goal of the PR is to remove the need to parse XML properties files. Leaving the XML parsing in the code would imply that this PR is not needed,

Reply:
That goal is still valid, in the greater time frame of the project.  But replacing XML with system properties is a purely lateral move, so if the change were just that, it would not be worthwhile.

If support for XML configuration were considered unimportant enough that we would discontinue it without deprecation (this would be a question for the project team, I'm not sure exactly what the criteria are), then I would just drop it with no replacement and let the squeaky wheels squeak.  TBH this is in many ways the best option.

If, however, I am right in my guess that we can't just drop support for it due to compatibility requirements, then chronologically, the plan would necessarily be something like this:

 * Deprecate XML configuration (with a warning)
 * Introduce any missing, desired configuration options as requests are raised
 * Drop the XML configuration when all of the following occur:
    1. The volume of requests has dropped off sufficiently to allay worries of lost functionality
    2. We reach a sufficient compatibility milestone (2.0 perhaps, depends on the outcome of the ongoing deprecation process discussion)

But which course should be taken is not something I can answer directly.

Reply:
Okay